stablehlo/dialect/VhloBytecode.cpp

```diff
@@ -696,6 +927,185 @@ void VhloBytecodeInterface::write(TypeExtensionsAttr attr,
   writer.writeSignedVarInts(attr.getBounds());
 }
 
+//===----------------------------------------------------------------------===//
+// Forked Attributes
+//===----------------------------------------------------------------------===//
+
+namespace {
+template <typename AttrOrType>
+bool assertFromVhlo(AttrOrType val) {
```

Comment:
I wonder if we can check this via verifiers instead - both for attributes and types. This would allow for much better error messages.

Reply:
I like this idea, will try it out.

Reply:
Still working on this, but I don't think it blocks re-review of the other bits. I'm thinking of how to make this elegant.